packages/desktop-client/src/components/modals/CategoryMenu.tsx

```diff
@@ -0,0 +1,162 @@
+
+  function _onClose() {
+    modalProps?.onClose();
+    onClose?.();
+  }
+
+  function _onRename(newName) {
+    if (newName !== name) {
+      onSave?.({
+        ...category,
+        name: newName,
+      });
+    }
+  }
+
+  function _onToggleVisibility() {
+    onSave?.({
+      ...category,
+      hidden: !!!category.hidden,
+    });
+    _onClose();
+  }
+
+  function _onEditNotes() {
+    onEditNotes?.(id);
+  }
+
+  function _onDelete() {
+    onDelete?.(id);
+  }
+
+  function onNameChange(newName) {
+    _onRename(newName);
+    setName(newName);
+  }
+
+  const buttonStyle: CSSProperties = {
+    ...styles.mediumText,
+    flexBasis: '50%',
+    height: BUTTON_HEIGHT,
+  };
+
+  return (
+    <Modal
+      title={name}
+      titleStyle={styles.underlinedText}
+      showHeader={true}
+      focusAfterClose={false}
+      {...modalProps}
+      onClose={_onClose}
+      padding={0}
+      style={{
+        flex: 1,
+        height: '50vh',
+        padding: '0 10px',
+        borderRadius: '6px',
+      }}
+      editableTitle={true}
+      onTitleChange={onNameChange}
+    >
+      {() => (
+        <View
+          style={{
+            flex: 1,
+            flexDirection: 'column',
+          }}
+        >
+          <View
+            style={{
+              overflowY: 'auto',
+              flex: 1,
+            }}
+          >
+            <Notes
+              notes={originalNotes?.length > 0 ? originalNotes : 'No notes'}
+              editable={false}
+              focused={false}
+              getStyle={editable => ({
+                borderRadius: 6,
+                ...((!originalNotes || originalNotes.length === 0) && {
+                  justifySelf: 'center',
+                  alignSelf: 'center',
+                  color: theme.pageTextSubdued,
+                }),
+              })}
+            />
+          </View>
+          <View
+            style={{
+              flexDirection: 'row',
+              flexWrap: 'wrap',
+              justifyContent: 'space-between',
+              alignContent: 'space-between',
+              paddingTop: 10,
+              paddingBottom: 10,
+            }}
+          >
+            <Button style={buttonStyle} onPointerUp={_onEditNotes}>
+              <NotesPaper width={20} height={20} style={{ paddingRight: 5 }} />
+              Edit notes
+            </Button>
+            <Button style={buttonStyle} onPointerUp={_onToggleVisibility}>
+              {category.hidden ? (
+                <ViewShow width={20} height={20} style={{ paddingRight: 5 }} />
+              ) : (
+                <ViewHide width={20} height={20} style={{ paddingRight: 5 }} />
+              )}
+              {category.hidden ? 'Unhide' : 'Hide'}
+            </Button>
+            <Button style={buttonStyle} onPointerUp={_onDelete}>
+              <Trash width={17} height={17} style={{ paddingRight: 5 }} />
+              Delete
+            </Button>
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏UI could be improved. Either a small spacing between the buttons should be added or perhaps we need to remove the rounded corners where they touch.

<img width="419" alt="Screenshot 2023-12-14 at 19 52 51" src="https://github.com/actualbudget/actual/assets/886567/8bb1530e-9f7f-41b8-93e9-aeb5fa0ae84f">